src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -726,6 +726,28 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                             throw ExceptionUtilities.UnexpectedValue(implicitIndexer.IndexerOrSliceAccess.Kind);
                     }
 
+                case BoundKind.InlineArrayAccess:
+                    {
+                        var elementAccess = (BoundInlineArrayAccess)expr;
+
+                        if (elementAccess.IsValue || elementAccess.GetItemOrSliceHelper is WellKnownMember.System_Span_T__Slice_Int_Int or WellKnownMember.System_ReadOnlySpan_T__Slice_Int_Int)
+                        {
+                            // Strict RValue
+                            break;
+                        }
+
+                        var getItemOrSliceHelper = (MethodSymbol)Compilation.GetWellKnownTypeMember(elementAccess.GetItemOrSliceHelper);
+
+                        if (getItemOrSliceHelper is null)
+                        {
+                            return true;
+                        }
+
+                        getItemOrSliceHelper = getItemOrSliceHelper.AsMember(getItemOrSliceHelper.ContainingType.Construct(ImmutableArray.Create(elementAccess.Expression.Type.TryGetInlineArrayElementType())));
```

Comment:
>TryGetInlineArrayElementType() [](http://example.com/codeflow?start=175&length=30)

Can this return `TypeWithAnnotations { HasType: false }`?

Reply:
> Can this return ```TypeWithAnnotations { HasType: false }```?

Not for receiver of ```BoundInlineArrayAccess```.